haystack/preprocessor/preprocessor.py

```diff
@@ -0,0 +1,205 @@
+import logging
+import re
+from copy import deepcopy
+from functools import partial, reduce
+from itertools import chain
+from typing import List, Optional, Tuple, Generator, Set
+
+import nltk
+from more_itertools import windowed
+
+from haystack.preprocessor.base import BasePreProcessor
+
+logger = logging.getLogger(__name__)
+
+
+class PreProcessor(BasePreProcessor):
+    def __init__(
+        self,
+        clean_whitespace: Optional[bool] = True,
+        clean_header_footer: Optional[bool] = False,
+        clean_empty_lines: Optional[bool] = True,
+        split_by: Optional[str] = "passage",
+        split_size: Optional[int] = 10,
+        split_stride: Optional[int] = None,
+        split_mid_sentence: Optional[bool] = True,
+    ):
+        """
+        :param clean_header_footer: use heuristic to remove footers and headers across different pages by searching
+                                     for the longest common string. This heuristic uses exact matches and therefore
+                                     works well for footers like "Copyright 2019 by XXX", but won't detect "Page 3 of 4"
+                                     or similar.
+        :param clean_whitespace: strip whitespaces before or after each line in the text.
+        :param clean_empty_lines: remove more than two empty lines in the text.
+        :param split_by: split the document by "word", "sentence", or "passage". Set to None to disable splitting.
+        :param split_size: n number of splits to merge as a single document. For instance, if n -> 10 & split_by ->
+                           "sentence", then each output document will have 10 sentences.
+        :param split_stride: overlap splits by "sliding window". Set to None to disable it.
```

Comment:
is it an integer of tokens or percentage of document ?

Reply:
Integer of the unit chosen in "split_by" => so can be num of words / sentences and later possibly also tokens